### ISSUE TITLE
🐛 Bug: Filter's visibility for different users

### ISSUE THREAD

**Issue by stepanof** [nocodb/nocodb#4228]
### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Steps to reproduce ?

Hello team!
Our team have faced with some bugs during working with table's filters.
Steps to reproduce:
1. Create new project, create new table like this (via **nocodb_user_01**)

![image](https://user-images.githubusercontent.com/68236577/198577811-f7384079-c12a-4607-ba7d-6381b6e4a050.png)
2. Invite **nocodb_user_02** to the project
3. Log in as **nocodb_user_02** in another browser or in Chrome Incognito mode (the goal is to have 2 active browser sessions with differrent users at the same moment)
4. Create Filter like this: (via **nocodb_user_02**)

![image](https://user-images.githubusercontent.com/68236577/198579068-3eeac88e-7d05-4db6-a2f8-fd3bfb45abf9.png)
**5. The first bug** is that filter doesn't apply automaticly even though the button 'Auto aplly' is on (via **nocodb_user_02**)
6. Turn off button 'Auto apply' and click 'Apply changes' (via **nocodb_user_02**)

![image](https://user-images.githubusercontent.com/68236577/198579689-a3a53237-c88a-42a5-ae52-c54383d932c2.png)

Filter is applied.

![image](https://user-images.githubusercontent.com/68236577/198582168-19de28da-50c4-4e84-8667-fa12faf24eb5.png)

7. Open browser with session as **nocodb_user_01**, refresh page and see filtered table. 
Add one more filter like this:

![image](https://user-images.githubusercontent.com/68236577/198582608-4178858d-c36b-451a-b674-6b6e63d3d542.png)

Click 'Apply changes'. 

**The second bug** is that I don't see any rows even though row with 'id = 2' is suitable for filter requirements.

![image](https://user-images.githubusercontent.com/68236577/198582982-79db95d7-b1da-4d94-82ff-8bc9d52f5b39.png)

8. Open browser with session as **nocodb_user_02**, refresh page and see filtered table. 
**The third bug** is that **nocodb_user_02** doesn't see any applied filters for that table.

![image](https://user-images.githubusercontent.com/68236577/198583300-b1639ddd-753d-4f7b-94b1-d1aa39497a29.png)

### Desired Behavior

1) button 'Auto aplly' must work
2) double filter for different column must work correctly
3) any user must see all applied filters

### Project Details

Node: **v12.22.12**
Arch: **x64**
Platform: **linux**
Docker: **true**
Database: **pg**
ProjectOnRootDB: **true**
RootDB: **pg**
PackageVersion: **0.98.1**

**Comment by dstala** [comment]
Worked as expected for me on chrome browser for below configuration

Node: **v16.14.0**
Arch: **arm64**
Platform: **darwin**
Docker: **false**
Database: **pg**
ProjectOnRootDB: **true**
RootDB: **pg**
PackageVersion: **0.98.3**

**Comment by stepanof** [comment]
@dstala I have upgraded nocodb to 0.93 and can't repeat the first and the third bug. But anyway when I apply 2 filters I don't see any rows
![image](https://user-images.githubusercontent.com/68236577/200610058-3afab569-f3da-487d-a7f4-a6c1e109e7ee.png)

**Comment by emre1e** [comment]
I think this is an issue in the roles & rights.
If you create a filter with the Creator Role, then users with the editor role cant see the filter anymore.

Change your nocodb_user_02 to creator and try to see it.

**Comment by stepanof** [comment]
@emre1e
@dstala
you are quite right! I felt difficultly to find out the specific the conditions which lead me to error and your suggestion helped me.
1) Editor can't see filters which were set by Creator
2) 'Auto apply' button for filter works only for Creator but doesn't work for user with Editor role

**Comment by RawmanGrey** [comment]
As of version 0.99.2 same filter behavior happens: editor can't change filter if it was touched by creator. This can go wild and not quite intuitive... 

I think, filters set by owner shouldn't be mutable by editors and they have to be shown in UI. But editor should have an option to add some conditions to filter. Or just keep it simple — leave filters common: let both owner and editor do whatever they like.

<details> 
  <summary>Confusion .gif </summary>
   
![Peek 2022-11-30 14-38](https://user-images.githubusercontent.com/61871501/204789402-f1cecc0e-2cf8-420c-a9e2-f0ba465ef9b2.gif)

</details>

**Comment by kudryk-enverus** [comment]
There's something definitely wrong going on. I've noticed the issues that @stepanof mentioned at first, and have just updated our deployment to 0.101.2 to ensure I've got the latest and greatest.

If the owner/super admin logs in and goes to a table where an editor user has set filters, the owner can see the filters, but then the editor users will no longer be able to see the filters they set.

What I'm also seeing, is if editor 1 sets a filter, that user can see a the filter and the resulting subset of results.

![Editor1](https://user-images.githubusercontent.com/82043488/213001098-0138aec7-7093-4e2c-a4cf-5e4eb2b77a3e.jpg)

But if editor 2 logs in and goes to the same table, they see the same subset of results, but cannot see the filter values set by user #2.

![Editor2](https://user-images.githubusercontent.com/82043488/213001259-c3aee97e-d2e8-49f9-bc42-0db0398c75e1.jpg)

It's as if [Personal View have been implemented](https://github.com/nocodb/nocodb/discussions/4860), but in a way that the first person to set the filter is the only one who can use the filter. Until personal views are implemented, I agree with @RawmanGrey suggestion of keeping things simple "let both owner and editor do whatever they like". I would add that changes by 1 user should be visible by all other users... until personal views are implemented.
